php_pthreads.c

```diff
@@ -156,11 +156,11 @@ static inline void pthreads_execute_ex(zend_execute_data *data) {
 /* {{{ */
 static inline zend_bool pthreads_verify_type(zend_execute_data *execute_data, zval *var, zend_arg_info *info) {
 	pthreads_object_t *threaded = NULL;
-
+        
 	if (!var ||
 #if PHP_VERSION_ID >= 70200
-		!ZEND_SAME_FAKE_TYPE(ZEND_TYPE_CODE(info->type), Z_TYPE_P(var)) &&
-		!ZEND_TYPE_IS_CLASS(info->type) ||
+                Z_ISNULL_P(var) ||
```

Comment:
feels like null check is wrong ...

it may break nullable types ... may be better to introduce null check with allow_null verification from arg info ...